### ISSUE TITLE
Getting "grep: illegal option -- o"

### ISSUE THREAD

**Issue by ghost** [arekinath/triton-dehydrated#19]
Hey there,

I recently updated to the latest version (1.2.3) and I'm getting grep errors. The whole process seems to succeed, but grep may not be doing what it is supposed to.

```[root@headnode (xxxx) ~]# /opt/letsencrypt/dehydrated -c
# INFO: Using main config file /opt/letsencrypt/config
startup_hook
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
Processing xxxx
 + Checking domain name(s) of existing cert... unchanged.
 + Checking expire date of existing cert...
 + Valid till Jul 31 10:32:47 2019 GMT (Longer than 30 days). Skipping renew!
unchanged_cert
OK: adminui certificate up to date
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
Processing xxxx
 + Checking domain name(s) of existing cert... unchanged.
 + Checking expire date of existing cert...
 + Valid till Jul 31 10:36:46 2019 GMT (Longer than 30 days). Skipping renew!
unchanged_cert
OK: cloudapi certificate up to date
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
Processing xxxx
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 1 authorizations URLs from the CA
 + Handling authorization for xxxx
 + 1 pending challenge(s)
 + Deploying challenge tokens...
deploy_challenge
Successfully updated VM yyyy
OK: deployed dns token for xxxx successfully
 + Responding to challenge for xxxx authorization...
 + Challenge is valid!
 + Cleaning challenge tokens...
clean_challenge
Successfully updated VM yyyy
 + Requesting certificate...
 + Checking certificate...
 + Done!
 + Creating fullchain.pem...
deploy_cert
Installing certificate
Restarting sdc-docker service
OK: docker certificate deployed
 + Done!
exit_hook
```

**Comment by bahamat** [comment]
We're aware of this error. It's a GNU specific option that isn't available on SmartOS, however it does still process and issue certs correctly, so we're not considering this a high priority right now.

**Comment by sreboot** [comment]
Can be fixed by installing GNU grep with `pkgin in grep-3.3` and removing `/usr/xpg4/bin` from this line https://github.com/joyent/triton-dehydrated/blob/master/dehydrated#L12

Have not investigated cases where `/usr/xpg4/bin` is required, but I am guessing it is used in the global zone for Triton specifics?

This grep error btw breaks wildcard certificate requests. @bahamat please, could you confirm whether `/usr/xpg4/bin` is required at all in the `$PATH` inside non global zones?

**Comment by bahamat** [comment]
We issue wildcard certs for cmon from the headnode global zone. This error doesn’t affect it.

Are you sure that it didn’t fail for a different reason?

**Comment by sreboot** [comment]
I am pretty sure it fails with this notion in domains.txt:
`*.service.example.com > star_service_example_com`

**Comment by bahamat** [comment]
Ah, got it.

First, you can work around this by having a SAN name instead of just a single name wildcard.

    service.example.com *.service.example.com

I'll look at removing `/usr/xpg4/bin`. I have a few other fixes that I'm working on today so I can try that as well.

**Comment by bahamat** [comment]
Fixed in f894cec269b29f6854569f38f5584517d2f0735d.

If gnu grep is installed from pkgsrc and is in the `$PATH` then it will be used. If it's not installed, it'll fall back to using the system grep.
